javax.mail.MessagingException: Could not convert socket to TLS;

We experiencing some strange problem with sending notifications to approvers. Here is the log:

```
2015-07-02T04:04:52.633927+00:00 app[web.1]: javax.mail.MessagingException: Could not convert socket to TLS;
2015-07-02T04:04:52.633931+00:00 app[web.1]:   nested exception is:
2015-07-02T04:04:52.641882+00:00 app[web.1]: 	javax.net.ssl.SSLException: java.lang.RuntimeException: Could not generate DH keypair
2015-07-02T04:04:52.652432+00:00 app[web.1]: 	at scalikejdbc.DB.localTx(DB.scala:75)
2015-07-02T04:04:52.652527+00:00 app[web.1]: 	at scalikejdbc.DB$$anonfun$localTx$1.apply(DB.scala:257)
2015-07-02T04:04:52.652921+00:00 app[web.1]: 	at scalikejdbc.DB$$anonfun$localTx$1.apply(DB.scala:256)
2015-07-02T04:04:52.653004+00:00 app[web.1]: 	at scalikejdbc.LoanPattern$class.using(LoanPattern.scala:33)
2015-07-02T04:04:52.653253+00:00 app[web.1]: 	at scalikejdbc.DB$.using(DB.scala:150)
2015-07-02T04:04:52.653674+00:00 app[web.1]: 	at controllers.Approving$$anonfun$notifyAllApprovers$1$$anonfun$apply$1.apply$mcV$sp(Approving.scala:62)
2015-07-02T04:04:52.653350+00:00 app[web.1]: 	at scalikejdbc.DB$.localTx(DB.scala:256)
2015-07-02T04:04:52.658647+00:00 app[web.1]: 	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
2015-07-02T04:04:52.657846+00:00 app[web.1]: 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
2015-07-02T04:04:52.647626+00:00 app[web.1]: 	at com.sun.mail.smtp.SMTPTransport.startTLS(SMTPTransport.java:1907)
2015-07-02T04:04:52.647735+00:00 app[web.1]: 	at com.sun.mail.smtp.SMTPTransport.protocolConnect(SMTPTransport.java:666)
2015-07-02T04:04:52.647907+00:00 app[web.1]: 	at javax.mail.Service.connect(Service.java:295)
2015-07-02T04:04:52.648207+00:00 app[web.1]: 	at javax.mail.Service.connect(Service.java:176)
2015-07-02T04:04:52.648349+00:00 app[web.1]: 	at javax.mail.Service.connect(Service.java:196)
2015-07-02T04:04:52.648580+00:00 app[web.1]: 	at javax.mail.Transport.send0(Transport.java:251)
2015-07-02T04:04:52.648838+00:00 app[web.1]: 	at javax.mail.Transport.send(Transport.java:174)
2015-07-02T04:04:52.648939+00:00 app[web.1]: 	at helpers.Notifications$.sendApprovalMessage(Notifications.scala:64)
2015-07-02T04:04:52.649108+00:00 app[web.1]: 	at helpers.Notifications$.notifyApproversAboutSuggestedQuote(Notifications.scala:23)
2015-07-02T04:04:52.649376+00:00 app[web.1]: 	at controllers.Approving$$anonfun$notifyAllApprovers$1$$anonfun$apply$1$$anonfun$1$$anonfun$apply$5.apply(Approving.scala:66)
2015-07-02T04:04:52.649549+00:00 app[web.1]: 	at controllers.Approving$$anonfun$notifyAllApprovers$1$$anonfun$apply$1$$anonfun$1$$anonfun$apply$5.apply(Approving.scala:65)
2015-07-02T04:04:52.649921+00:00 app[web.1]: 	at scala.collection.immutable.List.foreach(List.scala:381)
2015-07-02T04:04:52.650097+00:00 app[web.1]: 	at controllers.Approving$$anonfun$notifyAllApprovers$1$$anonfun$apply$1$$anonfun$1.apply(Approving.scala:65)
2015-07-02T04:04:52.650342+00:00 app[web.1]: 	at controllers.Approving$$anonfun$notifyAllApprovers$1$$anonfun$apply$1$$anonfun$1.apply(Approving.scala:62)
2015-07-02T04:04:52.650602+00:00 app[web.1]: 	at scalikejdbc.DBConnection$$anonfun$_localTx$1$1.apply(DBConnection.scala:252)
2015-07-02T04:04:52.650702+00:00 app[web.1]: 	at scala.util.control.Exception$Catch.apply(Exception.scala:103)
2015-07-02T04:04:52.650868+00:00 app[web.1]: 	at scalikejdbc.DBConnection$class._localTx$1(DBConnection.scala:250)
2015-07-02T04:04:52.651127+00:00 app[web.1]: 	at scalikejdbc.DBConnection$class.localTx(DBConnection.scala:258)
2015-07-02T04:04:52.656442+00:00 app[web.1]: 	at controllers.Approving$$anonfun$notifyAllApprovers$1$$anonfun$apply$1.apply(Approving.scala:62)
2015-07-02T04:04:52.656541+00:00 app[web.1]: 	at controllers.Approving$$anonfun$notifyAllApprovers$1$$anonfun$apply$1.apply(Approving.scala:62)
2015-07-02T04:04:52.656733+00:00 app[web.1]: 	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
2015-07-02T04:04:52.656989+00:00 app[web.1]: 	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
2015-07-02T04:04:52.657090+00:00 app[web.1]: 	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
2015-07-02T04:04:52.657472+00:00 app[web.1]: 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
2015-07-02T04:04:52.658336+00:00 app[web.1]: 	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
2015-07-02T04:04:52.678985+00:00 app[web.1]: Caused by: javax.net.ssl.SSLException: java.lang.RuntimeException: Could not generate DH keypair
2015-07-02T04:04:52.679103+00:00 app[web.1]: 	at sun.security.ssl.Alerts.getSSLException(Alerts.java:208)
2015-07-02T04:04:52.679202+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1884)
2015-07-02T04:04:52.679304+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1842)
2015-07-02T04:04:52.679384+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1825)
2015-07-02T04:04:52.679488+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1346)
2015-07-02T04:04:52.679593+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1323)
2015-07-02T04:04:52.679693+00:00 app[web.1]: 	at com.sun.mail.util.SocketFetcher.configureSSLSocket(SocketFetcher.java:527)
2015-07-02T04:04:52.679966+00:00 app[web.1]: 	at com.sun.mail.util.SocketFetcher.startTLS(SocketFetcher.java:464)
2015-07-02T04:04:52.680074+00:00 app[web.1]: 	at com.sun.mail.smtp.SMTPTransport.startTLS(SMTPTransport.java:1902)
2015-07-02T04:04:52.680294+00:00 app[web.1]: 	... 33 more
2015-07-02T04:04:52.684697+00:00 app[web.1]: Caused by: java.lang.RuntimeException: Could not generate DH keypair
2015-07-02T04:04:52.684927+00:00 app[web.1]: 	at sun.security.ssl.DHCrypt.<init>(DHCrypt.java:136)
2015-07-02T04:04:52.685184+00:00 app[web.1]: 	at sun.security.ssl.ClientHandshaker.serverKeyExchange(ClientHandshaker.java:621)
2015-07-02T04:04:52.685287+00:00 app[web.1]: 	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:205)
2015-07-02T04:04:52.685548+00:00 app[web.1]: 	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:868)
2015-07-02T04:04:52.685648+00:00 app[web.1]: 	at sun.security.ssl.Handshaker.process_record(Handshaker.java:804)
2015-07-02T04:04:52.685907+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1016)
2015-07-02T04:04:52.686008+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1312)
2015-07-02T04:04:52.686264+00:00 app[web.1]: 	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1339)
2015-07-02T04:04:52.686364+00:00 app[web.1]: 	... 37 more
2015-07-02T04:04:52.694793+00:00 app[web.1]: Caused by: java.security.InvalidAlgorithmParameterException: Prime size must be multiple of 64, and can only range from 512 to 1024 (inclusive)
2015-07-02T04:04:52.694922+00:00 app[web.1]: 	at com.sun.crypto.provider.DHKeyPairGenerator.initialize(DHKeyPairGenerator.java:120)
2015-07-02T04:04:52.695139+00:00 app[web.1]: 	at java.security.KeyPairGenerator$Delegate.initialize(KeyPairGenerator.java:658)
2015-07-02T04:04:52.695380+00:00 app[web.1]: 	at sun.security.ssl.DHCrypt.<init>(DHCrypt.java:127)
2015-07-02T04:04:52.695620+00:00 app[web.1]: 	... 44 more
```

I'll try too investigate it this evening.

--- a/system.properties
+++ b/system.properties
@@ -1,1 +1,1 @@
-java.runtime.version=1.7
+java.runtime.version=1.8

